bn256.go

```diff
@@ -105,6 +105,11 @@ func (e *G1) Marshal() []byte {
 	// Each value is a 256-bit number.
 	const numBytes = 256 / 8
 
+	if e.p == nil {
+		e.p = &curvePoint{}
+		e.p.SetInfinity()
+	}
+
```

Comment:
This isn't necessary on G1 or G2 because of the MakeAffine below

Reply:
Right, now is fixed.